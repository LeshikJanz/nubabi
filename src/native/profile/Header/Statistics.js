// @flow
import React from 'react';
import { View, Text, StyleSheet } from 'react-native';

type Props = {
  achievements: number,
  favourites: number,
};

const Statistics = ({ achievements, favourites }: Props) => {
  return (
    <View style={styles.stats}>
      <View style={styles.achievements}>
        <Text style={styles.achievementsHeader}>ACHIEVEMENTS</Text>
        <Text style={styles.achievementsValue}>
          {achievements}
        </Text>
      </View>
      <View style={styles.favourites}>
        <Text style={styles.favouritesHeader}>FAVOURITES</Text>
        <Text style={styles.favouritesValue}>
          {favourites}
        </Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  stats: {
    backgroundColor: 'transparent',
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    marginTop: -5,
  },
  achievements: {
    flex: 2,
    alignItems: 'center',
    borderRightWidth: 1,
    borderColor: 'rgba(255,255,255,0.2)',
  },
  achievementsHeader: {
    color: '#fff',
    fontSize: 12,
    fontWeight: '700',
  },
  achievementsValue: {
    color: '#fff',
    fontSize: 18,
  },
  favourites: {
    flex: 2,
    alignItems: 'center',
  },
  favouritesHeader: {
    color: '#fff',
    fontSize: 12,
    fontWeight: '700',
  },
  favouritesValue: {
    color: '#fff',
    fontSize: 18,
  },
});

export default Statistics;
