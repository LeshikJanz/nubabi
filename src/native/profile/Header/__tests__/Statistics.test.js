import 'react-native';
import React from 'react';
import Statistics from '../Statistics';
import renderer from 'react-test-renderer';

test('it renders correctly', () => {
  const tree = renderer
    .create(<Statistics achievements={5} favourites={10} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
