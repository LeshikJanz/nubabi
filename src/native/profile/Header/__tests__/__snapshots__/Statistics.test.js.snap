// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "flex": 1,
      "flexDirection": "row",
      "marginTop": -5,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "rgba(255,255,255,0.2)",
        "borderRightWidth": 1,
        "flex": 2,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontSize": 12,
          "fontWeight": "700",
        }
      }
    >
      ACHIEVEMENTS
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontSize": 18,
        }
      }
    >
      5
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 2,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontSize": 12,
          "fontWeight": "700",
        }
      }
    >
      FAVOURITES
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontSize": 18,
        }
      }
    >
      10
    </Text>
  </View>
</View>
`;
